Capstone - Part 1

Topic: Initial Data Exploration

Created By: Jason

In [1]:
#Mounting our Drive to Google Colab
from google.colab import drive
drive.mount('/drive')

Mounted at /drive


In [2]:
# Import necessary libraries

import pandas as pd # data analysis
import numpy as np # arrays

In [3]:
path = '/drive/MyDrive/Colab Notebooks/CLEAN_2_Tour Agencies_v2.csv'
data = pd.read_csv(path)

Initial Data Exploration

In [4]:
# View all variables in the dataset
data.columns

Index(['I have read and understood the above information. I understand that, because my answers will be fully anonymized, it will not be possible to withdraw them from the study once I have completed the survey. I agree to take part in this questionnaire survey a',
       'Are you currently residing in Malaysia?', 'gender', 'age',
       'household_income', 'marital_status', 'chidlren_no', 'education_level',
       'occupation', 'trip_plan_likelihood', 'travel_purpose',
       'source_travel_info', 'pref_socialmed_info', 'pref_transport_mode',
       'pref_length_stay', 'travel_pay_willingness', 'pref_poi',
       'pref_group_size', 'pref_travel_destination'],
      dtype='object')

In [5]:
# Remove irrelavant columns 
# First 2 variables will not be useful for further analysis (just for questionnaire survey purposes)

data = data.drop(['I have read and understood the above information. I understand that, because my answers will be fully anonymized, it will not be possible to withdraw them from the study once I have completed the survey. I agree to take part in this questionnaire survey a', 
                  'Are you currently residing in Malaysia?'], axis=1)
data.columns 

Index(['gender', 'age', 'household_income', 'marital_status', 'chidlren_no',
       'education_level', 'occupation', 'trip_plan_likelihood',
       'travel_purpose', 'source_travel_info', 'pref_socialmed_info',
       'pref_transport_mode', 'pref_length_stay', 'travel_pay_willingness',
       'pref_poi', 'pref_group_size', 'pref_travel_destination'],
      dtype='object')

In [6]:
# View dataset total observations and variables
data.shape

(115, 17)

In [7]:
# Check the variable information (Data type etc.)
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115 entries, 0 to 114
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   gender                   115 non-null    object
 1   age                      115 non-null    object
 2   household_income         115 non-null    object
 3   marital_status           115 non-null    object
 4   chidlren_no              115 non-null    object
 5   education_level          115 non-null    object
 6   occupation               115 non-null    object
 7   trip_plan_likelihood     115 non-null    int64 
 8   travel_purpose           115 non-null    object
 9   source_travel_info       115 non-null    object
 10  pref_socialmed_info      115 non-null    object
 11  pref_transport_mode      115 non-null    object
 12  pref_length_stay         115 non-null    object
 13  travel_pay_willingness   115 non-null    object
 14  pref_poi                 115 non-null    o

In [8]:
# Simple view on the dataset 
data.head()

gender                age    household_income marital_status chidlren_no  \
0  Female  18 - 25 years old    More than 10,959         Single        None   
1    Male  18 - 25 years old    More than 10,959         Single        None   
2  Female  18 - 25 years old  Less than RM 4,850         Single        None   
3    Male  18 - 25 years old  Less than RM 4,850         Single        None   
4  Female  18 - 25 years old  Less than RM 4,850         Single        None   

                     education_level occupation  trip_plan_likelihood  \
0                    Master's Degree    Student                     5   
1  Undergraduate / Bachelor's Degree   Employed                     4   
2  Undergraduate / Bachelor's Degree   Employed                     4   
3                    Master's Degree   Employed                     5   
4                    Master's Degree   Employed                     5   

      travel_purpose     source_travel_info pref_socialmed_info  \
0        Family Trip           Social media             YouTube   
1  Holiday & Leisure  Friends and relatives             YouTube   
2  Holiday & Leisure           Social media             YouTube   
3  Holiday & Leisure           Social media         Travel Blog   
4  Holiday & Leisure           Social media         Travel Blog   

  pref_transport_mode  pref_length_stay travel_pay_willingness  \
0              Flight  Less than 3 days   Between RM 200 – 300   
1              Flight            3 Days       Less than RM 200   
2              Flight            3 Days       More than RM 300   
3               Train  Less than 3 days   Between RM 200 – 300   
4              Flight            3 Days       More than RM 300   

                                            pref_poi  \
0                        Cultural or Historical Site   
1                        Cultural or Historical Site   
2                              Beaches and waterfall   
3                              Beaches and waterfall   
4  Adventure and Activity (Ex: Hiking, Mountain C...   

                pref_group_size  \
0  Family Travel (3 – 6 people)   
1  Family Travel (3 – 6 people)   
2   Small Group (6 – 10 people)   
3  Family Travel (3 – 6 people)   
4  Family Travel (3 – 6 people)   

                             pref_travel_destination  
0                  East Malaysia (Sabah and Sarawak)  
1  Southern Region (Malacca, Johor and Negeri Sem...  
2  East Coast Region (Kelantan, Terengganu and Pa...  
3  East Coast Region (Kelantan, Terengganu and Pa...  
4  East Coast Region (Kelantan, Terengganu and Pa...

Conclusion:

1.  A total of 115 observations
2.  17 variables involved in the dataset after the removal of 2 variables
3.  There are no missing values in the dataset
4.  16 variables are type "string" and only 1 variable is type "integer"


